### PR TITLE
docs(saas): data-retention

### PR DESCRIPTION
**Pull request by Sijoma**
## What is the purpose of the change

This adds the documentation for the data retention concepts in SaaS. It would be great if someone could check whether I did all the necessary work - this is my first time contributing here. I tried to stick to the [review process](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process).

@felix-mueller @menski I'm not sure if there needs to be more information about this topic in general. Feel free to add things / mention what should be explained more. 

@christinaausley I would be super happy for any kind of suggestions and improvements. I'm not familiar with our wording etc. so If I use any incorrect terms let me know. 

## Are there related marketing activities

_Are there any marketing activities related to the related feature or component? If so, please briefly describe them._

## When should this change go live?

_Does this change need to be released before a certain date or milestone? If so, when?_

This should be added for `8.1 and future` (https://github.com/camunda/product-hub/issues/167)

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

relates to Add data retention documentation #1460 #1032

### CODE REVIEW

File: docs/components/concepts/data-retention.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+id: data-retention
+title: "Data retention"
+---
+
+In Camunda Platform 8 SaaS, the following data retention strategies are implemented. This is necessary as the amount of data can grow significantly over time. These settings are a balance between performance and usability.
+
+## Retention time of each application
+
+The following time-to-live settings are configured in SaaS for each application.
```
Comment (felix-mueller) [inline docs/components/concepts/data-retention.md:10]: Could we add three things:
- These are defaults, please reach out if you have specific requirements
- These are defaults for Enterprise Plan
- Data Retention can be configured for Self-Managed.

Comment (Sijoma) [reply docs/components/concepts/data-retention.md:10]: Thanks for the input. The defaults for an enterprise plan are the same right?

Comment (felix-mueller) [reply docs/components/concepts/data-retention.md:10]: I meant that I think we should for now call them "defaults for Enterprise" as we still want to consider changing defaults for Prof and Trial. So we can add later add information about Prof and Trial

Comment (Sijoma) [reply docs/components/concepts/data-retention.md]: Added the mentioned things.

**Review by felix-mueller — DISMISSED** [review]
lgtm now thank you!!!

**Review by felix-mueller — APPROVED** [review]

**Review by christinaausley — APPROVED** [review]
Nice 👍 . I made a few technical adjustments in my latest commit, including:
- Removal of "please".
- A link to the mentioned Self-Managed section.
- Tabs for links.
- Added description in header.
Let me know if you have any questions about these changes.

**Pull request merged**